app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -16,16 +16,21 @@ function ContainerTopologyCtrl($scope, $http, $interval, topologyService, $windo
   $scope.d3 = d3;
 
   $scope.refresh = function() {
-    var id;
+    var id, type;
     var pathname = $window.location.pathname.replace(/\/$/, '');
     if (pathname.match(/show$/)) {
       id = '';
+    } else if (pathname.match('/(.+)/show/([0-9]+)')) {
+      var arr = pathname.match('/(.+)/show/([0-9]+)')
+      type = arr[1] + '_topology'
+      id = '/' + arr[2]
     } else {
       // search for pattern ^/<controler>/<id>$ in the pathname
+      type = 'container_topology'
```

Comment:
> Correct me if I'm wrong please, but .. isn't it always container_topology here? We don't really reuse the controller for any other topologies now, do we?

in this case it could also be `container_project_topology`. `container_topology` would lead to code from `ContainerTopologyService` which generates topology for `ems` and not projects.




> @zeari @martinpovolny this logic of this functions seems to become a little complex to maintain. Maybe we don't have to keep the old container_topology path if it's becoming an exception (as you need an extra else branch).

in that case we would need the `/ems_container/` path to build topology for `ems` .
right now there are 3 cases for the url:
* all ems topology (the path`/container_topology/` without an id)
* single ems topology (the same but with id)
* in this PR: topology for any entity: `'/container_project/' --> 'container_project_topology'`